### PR TITLE
I'm really bored of the hoisters in Fabricord

### PR DESCRIPTION
**Pull request by LambdAurora**
And multiple people already suggested multiple times to rename the hoisters.

This strategy is inspired from the strategy used on Jellysquid's Discord.

### CODE REVIEW

**Comment by i509VCB** [comment]
I will give this a look

**Comment by gdude2002** [comment]
This will conflict with [the nickname stuff in the Infractions extension](https://github.com/FabricMC/fabric-discord-bot/blob/kord-0.7/src/main/kotlin/net/fabricmc/bot/extensions/InfractionsExtension.kt#L855) as nick-locked users could cause some nickname looping if a staff member uses the `?nick` command to nick someone to a bad nickname.

File: src/main/kotlin/net/fabricmc/bot/extensions/NicknameExtension.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        event<MemberUpdateEvent> {
+            check(inGuild(config.getGuild()))
+            check { if (it.old != null) NAME_REGEX.matches(it.old!!.displayName) else false }
+
+            action { unhoistUser(event.member) }
+        }
+    }
+
+    /**
+     * Renames the member to "Z ; dumb hoist name".
```
Comment (mounderfod) [inline src/main/kotlin/net/fabricmc/bot/extensions/NicknameExtension.kt:49]: I do not think we should send them to the bottom. This is moderation, not trolling.

File: src/main/kotlin/net/fabricmc/bot/extensions/NicknameExtension.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import dev.kord.core.entity.Member
+import dev.kord.core.event.gateway.ReadyEvent
+import dev.kord.core.event.guild.MemberJoinEvent
+import dev.kord.core.event.guild.MemberUpdateEvent
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import net.fabricmc.bot.conf.config
+
+private val NAME_REGEX = Regex("^[!.\"+(-]")
```
Comment (mounderfod) [inline src/main/kotlin/net/fabricmc/bot/extensions/NicknameExtension.kt:16]: Aren't there other characters which are not listed here which can hoist, like `%`?

**Review by mounderfod — CHANGES_REQUESTED** [review]

**Pull request closed without merging**